github.go

```diff
@@ -202,6 +207,29 @@ func createGitHubRepoUrlForPath(repo GitHubRepo, path string) string {
 	return fmt.Sprintf("repos/%s/%s/%s", repo.Owner, repo.Name, path)
 }
 
+// Get the Next paginated path from the link url api.github.com/repos/:owner/:repo/:path
+// Links are formatted: "<url>; rel=next, <url>; rel=last"
+func getNextPath(links string) string {
+	if len(links) == 0 {
+		return ""
+	}
+
+	for _, link := range strings.Split(links, ",") {
+
+		if attrs := strings.Split(link, ";"); len(attrs) > 1 {
+			url, rel := attrs[0], attrs[1]
+
+			// Get only the next link
+			if strings.Contains(rel, "next") {
+				url = strings.Trim(url, " <>")
+				return url[strings.LastIndex(url, "/")+1:]
+			}
+		}
+	}
```

Comment:
There are a number of corner cases here that could cause issues: e.g., some new `rel` type that happens to contain the word "next"; handling of whitespace; index math; etc.

Perhaps a regex would be a safer way to parse this? That is, split on comma, and then use a regex like `(.+?);\\s*rel=next`... And get the URL out of the capture group.

Reply:
Also, whatever implementation you use, could you add some table-driven unit tests for `getNextPath`?

Reply:
I agree that a regex would be a safer way to parse this. I'll make those changes and add tests for this function. Thank you for the suggestions.